src/globus_sdk/experimental/login_flow_manager/command_line_login_flow_manager.py

```diff
@@ -0,0 +1,79 @@
+from globus_sdk import AuthLoginClient, OAuthTokenResponse
+from globus_sdk.experimental.auth_requirements_error import (
+    GlobusAuthorizationParameters,
+)
+from globus_sdk.experimental.login_flow_manager import LoginFlowManager
+
+
+class CommandLineLoginFlowManager(LoginFlowManager):
+    """
+    A ``CommandLineLoginFlowManager`` is a ``LoginFlowManager`` that uses the command
+    line for interacting with the user during its interactive login flows.
+    """
+
+    def __init__(
+        self,
+        login_client: AuthLoginClient,
+        *,
+        login_prompt: str = "Please authenticate with Globus here",
+        code_prompt: str = "Enter the resulting Authorization Code here",
+    ):
+        """
+        :param login_client: The ``AuthLoginClient`` that will be making the Globus
+            Auth API calls needed for the authentication flow.
+        :param login_prompt: The string that will be output to the command line
+            prompting the user to authenticate.
+        :param code_prompt: The string that will be output to the command line
+            prompting the user to enter their authorization code.
+        """
+        self.login_prompt = login_prompt
+        self.code_prompt = code_prompt
+        super().__init__(login_client)
+
+    def run_login_flow(
+        self,
+        auth_parameters: GlobusAuthorizationParameters,
+        *,
+        refresh_tokens: bool = False,
+    ) -> OAuthTokenResponse:
+        """
+        Run an interactive login flow on the command line to get tokens for the user.
+
+        :param auth_parameters: ``GlobusAuthorizationParameters`` passed through
+            to the authentication flow to control how the user will authenticate.
+        :param refresh_tokens: Whether or not refresh tokens will be requested.
+        """
+
+        # type is ignored here as AuthLoginClient does not provide a signature for
+        # oauth2_start_flow since it has different positional arguments between
+        # NativeAppAuthClient and ConfidentialAppAuthClient
+        self.login_client.oauth2_start_flow(  # type: ignore
```

Comment:
This looks fine to me. It's wholly self-contained in this method and looks like it works just fine.

Reply:
I agree that this seems fine -- for this code, as a user of the client classes.

It strikes me as a "code/design smell in the clients themselves". I'd love to resolve it, but I think it's safely out of scope for this PR to even try to fix that issue. I'm not sure what the fix really is. (A new method on the base accepting `*args` and raising notimplemented? :thinking: )